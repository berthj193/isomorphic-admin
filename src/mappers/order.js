import _map from 'lodash/map';

export const toListItem = orders =>
  _map(orders, order => {
    const { customer } = order;

    return {
      key: order.id,
      id: order.id,
      email: customer.email,
      status: 'â€“',
      customer_name: `${customer.given_name} ${customer.family_name}`,
      grand_total: order.grand_total.value,
      created_at: order.created_at,
    };
  });

