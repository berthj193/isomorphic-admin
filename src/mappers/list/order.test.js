import { toListItem } from '../order';

describe('Order mapper', () => {
  describe('toListItem()', () => {
    it('should flatten order object', () => {
      const data = {
        id: 'My0rd3r',
        customer: {
          email: 'my@email.com',
          given_name: 'Miroslaw',
          family_name: 'Yarzyna',
        },
        grand_total: {
          value: '1.000.000',
        },
        created_at: '2018-07-03T12:24:06+00:00',
      };
      expect(toListItem([data])).toMatchSnapshot();
    });

    it('should skip unpredicted fields', () => {
      const data = {
        id: 'My0rd3r',
        customer: {
          email: 'my@email.com',
          given_name: 'Miroslaw',
          family_name: 'Yarzyna',
        },
        grand_total: {
          value: '1.000.000',
        },
        created_at: '2018-07-03T12:24:06+00:00',
        iShould: 'BeHere',
      };
      expect(toListItem([data])).toMatchSnapshot();
    });
  });
});
