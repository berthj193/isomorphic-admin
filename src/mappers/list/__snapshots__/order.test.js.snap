// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order mapper toListItem() should flatten order object 1`] = `
Array [
  Object {
    "created_at": "2018-07-03T12:24:06+00:00",
    "customer_name": "Miroslaw Yarzyna",
    "email": "my@email.com",
    "grand_total": "1.000.000",
    "id": "My0rd3r",
    "key": "My0rd3r",
    "status": "–",
  },
]
`;

exports[`Order mapper toListItem() should skip unpredicted fields 1`] = `
Array [
  Object {
    "created_at": "2018-07-03T12:24:06+00:00",
    "customer_name": "Miroslaw Yarzyna",
    "email": "my@email.com",
    "grand_total": "1.000.000",
    "id": "My0rd3r",
    "key": "My0rd3r",
    "status": "–",
  },
]
`;
